### PR TITLE
Fix AI's MCV deploy issue for other locomotors

### LINKED ISSUE #621: Let the MCV belonging to AI and with special locomotor can deploys normally when game start
### First check

- [X] The issue is not introduced by Phobos or any other engine extension, such as Ares, Kratos etc.
- [X] The issue wasn't fixed in the most recent version of Ares/Phobos yet.
- [X] I agree to elaborate the details if requested and provide thorough testing if the bugfix is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, gifs, videos, debug and exception logs, etc)*.

### Description

In WW's original program, AI's MCV will go through a Hunt mission, If MCV's Locomotor uses Jumpjet or Chrono, AI will get stuck in this step, it will not deploy it as a constructionyard at the beginning of the game

### Conditions to reproduce

_No response_

### INI code

_No response_

### Steps to reproduce

Change the locomotor of MCV to jumpjet, etc

### Expected behaviour

AI deploys MCV as usual, no matter what locmotor it uses

### Actual behaviour

AI's MCV stands blankly at the birth point unless it is violently interrupted by something else, such as EMP

### Additional context

After using DynamicPatcher Script to replace hunt with guard, this problem has been solved. At present, no side effects have been found

### PR DESCRIPTION
**Pull request by chaserli**
Attempt to fix #621

Override AI's MCV's hunt mission by queueing a guard mission
MCV with chrono, hover or jumpjet locomotor should be able to deploy itself at the beginning of a skirmish game

### CODE REVIEW

**Comment by chaserli** [comment]
ttt4
If this is what he wanted

**Comment by DeathFishAtEase** [comment]
Jumpjet still has some problems finding open areas. If Chrono MCV can't deploy at the beginning, it will still stand as silly as before, even if the obstacles have been cleared
Non still picture

**Comment by chaserli** [comment]
> Jumpjet still has some problems finding open areas. If Chrono MCV can't deploy at the beginning, it will still stand as silly as before, even if the obstacles have been cleared Non still picture

What was the mission state of your hovering mcv? Hunt or Unload?

你能把地形放大一点吗，或者把地图和设置给我一下，我不知道咋复现这个

**Comment by DeathFishAtEase** [comment]
This map is provided by YR CNCNET Client and its file name is 8_malaboo_bay.map

**Comment by chaserli** [comment]
There are more bugs introduced. I never expected this to be that difficult. will work on it later

**Comment by chaserli** [comment]
@DeathFishAtEase it should looks better this time

**Comment by chaserli** [comment]
This one is also tested by @DeathFishAtEase 
Needs review from @Starkku since I was working inside his hook

**Comment by Belonit** [comment]
@chaserli It seems this PR is broken. I think you should create a new one instead.

File: src/Misc/Hooks.BugFixes.cpp [hunk 1]
```diff
@@ -507,6 +507,16 @@ DEFINE_HOOK(0x73EFD8, UnitClass_Mission_Hunt_DeploysInto, 0x6)
...
+	//Force guard mission to mcv when hunt
+	if (pThis->MissionStatus == 1)
+	{
+		if (auto conyType = pThis->Type->DeploysInto)
+		{
+			if (conyType->ConstructionYard || conyType->WeaponsFactory)
+				pThis->QueueMission(Mission::Guard, false);
+		}
+	}
+
```
Comment (Belonit) [inline src/Misc/Hooks.BugFixes.cpp:519]: If you do not plan to make new commits in this PR. You can take this suggestion to fix linter issues:
```suggestion

```

File: src/Misc/Hooks.BugFixes.cpp [hunk 2]
```diff
@@ -507,6 +507,16 @@ DEFINE_HOOK(0x73EFD8, UnitClass_Mission_Hunt_DeploysInto, 0x6)
 		return SkipToDeploy;
 	}
 
+	//Force guard mission to mcv when hunt
```
Comment (Belonit) [inline src/Misc/Hooks.BugFixes.cpp:510]: I'd rather add a space here
```suggestion
	// Force guard mission to mcv when hunt
```

**Review by Belonit — CHANGES_REQUESTED** [review]

**Pull request closed without merging**